geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/Bytes2ObjectOpenHashMap.java

```diff
@@ -0,0 +1,1287 @@
+        return size;
+      }
+
+      @Override
+      public boolean contains(Object v) {
+        return containsValue(v);
+      }
+
+      @Override
+      public void clear() {
+        Bytes2ObjectOpenHashMap.this.clear();
+      }
+    };
+  }
+
+  /**
+   * Rehashes the map, making the table as small as possible.
+   *
+   * <p>
+   * This method rehashes the table to the smallest size satisfying the
+   * load factor. It can be used when the set will not be changed anymore, so
+   * to optimize access speed and size.
+   *
+   * <p>
+   * If the table size is already the minimum possible, this method
+   * does nothing.
+   *
+   * @return true if there was enough memory to trim the map.
+   * @see #trim(int)
+   */
+  @SuppressWarnings("unused")
+  public boolean trim() {
+    return trim(size);
+  }
+
+  /**
+   * Rehashes this map if the table is too large.
+   *
+   * <p>
+   * Let <var>N</var> be the smallest table size that can hold
+   * <code>max(n,{@link #size()})</code> entries, still satisfying the load factor. If the current
+   * table size is smaller than or equal to <var>N</var>, this method does
+   * nothing. Otherwise, it rehashes this map in a table of size
+   * <var>N</var>.
+   *
+   * <p>
+   * This method is useful when reusing maps. {@linkplain #clear() Clearing a
+   * map} leaves the table size untouched. If you are reusing a map
+   * many times, you can call this method with a typical
+   * size to avoid keeping around a very large table just
+   * because of a few large transient maps.
+   *
+   * @param n the threshold for the trimming.
+   * @return true if there was enough memory to trim the map.
+   * @see #trim()
+   */
+  public boolean trim(final int n) {
+    final int l = HashCommon.nextPowerOfTwo((int) Math.ceil(n / getLoadFactor()));
+    if (l >= this.n || size > maxFill(l, getLoadFactor())) {
+      return true;
+    }
+    try {
+      rehash(l);
+    } catch (OutOfMemoryError cantDoIt) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Rehashes the map.
+   *
+   * <p>
+   * This method implements the basic rehashing strategy, and may be
+   * overridden by subclasses implementing different rehashing strategies (e.g.,
+   * disk-based rehashing). However, you should not override this method
+   * unless you understand the internal workings of this class.
+   *
+   * @param newN the new size
+   */
+  @SuppressWarnings("unchecked")
+  protected void rehash(final int newN) {
+    final byte[][] key = this.key;
+    final V[] value = this.value;
+    final int mask = newN - 1; // Note that this is used by the hashing macro
+    final byte[][] newKey = new byte[newN + 1][];
+    final V[] newValue = (V[]) new Object[newN + 1];
+    int i = n, pos;
+    for (int j = size(); j-- != 0;) {
+      // noinspection StatementWithEmptyBody
+      while (key[--i] == null) {
+      }
+      if (!(newKey[pos =
+          HashCommon.mix(strategy().hashCode(key[i])) & mask] == null)) {
```

Comment:
this can be changed to `if(newKey[pos=HashCommon.mix(strategy().hashCode(key[i])) & mask != null){`
same with the while below this